DPMMA-2550 Point Groups API Docs (#185)
mautic/developer-documentation-new: docs/rest_api/contacts.rst

       "success": true
   }

.. vale off

> Get All Contact Point Groups Scores

Reviewer: ```suggestion
Get all Contact Point Groups scores
```
We only need to capitalise the first word and any Mautic-specific terms

***********************************

.. vale on

> Get all Point Group scores for a specific contact.

Reviewer: ```suggestion
Get all Point Group scores for a specific Contact.
```

.. vale off

**HTTP Request**

.. vale on

``GET /api/contacts/{leadId}/points/groups``

**Response**

``Expected Response Code: 200``

.. code-block:: json

    {
        "total": 1,
        "groupScores": [
            {
                "score": 21,
                "group": {
                    "id": 6,
                    "name": "A"
                }
            }
        ]
    }

> The response will contain a list of Point Group scores for the Contact.

Reviewer: ```suggestion
The response contains a list of Point Group scores for the Contact.
```

Vale fix

.. vale off

> Get Single Contact Point Group Score

Reviewer: ```suggestion
Get single Contact Point Group score
```

************************************

.. vale on

> Get the score for a specific Point Group for a contact.

Reviewer: ```suggestion
Get the score for a specific Point Group for a Contact.
```

.. vale off

**HTTP Request**

.. vale on

``GET /api/contacts/{leadId}/points/groups/{groupId}``

**Response**

``Expected Response Code: 200``

.. code-block:: json

    {
        "groupScore": {
            "score": 21,
            "group": {
                "id": 6,
                "name": "A"
            }
        }
    }

> The response will contain the score for the specified Point Group for the Contact.

Reviewer: ```suggestion
The response contains the score for the specified Point Group for the Contact.
```
Vale fix

.. vale off

Add Contact Point Group Score
*****************************

.. vale on

> Add points to a specific Point Group for a contact.

Reviewer: ```suggestion
Add Points to a specific Point Group for a Contact.
```

.. vale off

**HTTP Request**

.. vale on

``POST /api/contacts/{leadId}/points/groups/{groupId}/plus/{value}``

**Data Parameters (optional)**

.. list-table::
   :header-rows: 1

   * - Name
     - Description
   * - ``eventname``
     - Name of the event
   * - ``actionname``
     - Name of the action

**Response**

``Expected Response Code: 200``

.. code-block:: json

    {
        "groupScore": {
            "score": 21,
            "group": {
                "id": 6,
                "name": "A"
            }
        }
    }

> The response will indicate the success of adding points to the specified Point Group for the Contact.

Reviewer: ```suggestion
The response indicates the success of adding Points to the specified Point Group for the Contact.
```

Vale fix

.. vale off

> Subtract Contact Point Group Score

Reviewer: ```suggestion
Subtract Contact Point Group score
```

**********************************

.. vale on

> Subtract points from a specific Point Group for a contact.

Reviewer: ```suggestion
Subtract Points from a specific Point Group for a Contact.
```

.. vale off

**HTTP Request**

.. vale on

``POST /api/contacts/{leadId}/points/groups/{groupId}/minus/{value}``

**Data Parameters (optional)**

.. list-table::
   :header-rows: 1

   * - Name
     - Description
   * - ``eventname``
     - Name of the event
   * - ``actionname``
     - Name of the action

**Response**

``Expected Response Code: 200``

.. code-block:: json

    {
        "groupScore": {
            "score": 21,
            "group": {
                "id": 6,
                "name": "A"
            }
        }
    }

> The response will indicate the success of subtracting points from the specified Point Group for the Contact.

Reviewer: ```suggestion
The response indicates the success of subtracting Points from the specified Point Group for the Contact.
```

.. vale off

Set Contact Point Group Score
*****************************

.. vale on

> Set the points for a specific Point Group for a contact.

Reviewer: ```suggestion
Set the Points for a specific Point Group for a Contact.
```

.. vale off

**HTTP Request**

.. vale on

``POST /api/contacts/{leadId}/points/groups/{groupId}/set/{value}``

**Data Parameters (optional)**

.. list-table::
   :header-rows: 1

   * - Name
     - Description
   * - ``eventname``
     - Name of the event
   * - ``actionname``
     - Name of the action

**Response**

``Expected Response Code: 200``

.. code-block:: json

    {
        "groupScore": {
            "score": 21,
            "group": {
                "id": 6,
                "name": "A"
            }
        }
    }

> The response will indicate the success of setting the points for the specified Point Group for the Contact.

Reviewer: ```suggestion
The response indicates the success of setting the Points for the specified Point Group for the Contact.
```

.. vale off

> Divide Contact Point Group Score

Reviewer: ```suggestion
Divide Contact Point Group score
```

********************************

.. vale on

> Divide the points of a specific Point Group for a contact by a value.

Reviewer: ```suggestion
Divide the Points of a specific Point Group for a Contact by a value.
```

.. vale off

**HTTP Request**

.. vale on

``POST /api/contacts/{leadId}/points/groups/{groupId}/divide/{value}``

**Data Parameters (optional)**

.. list-table::
   :header-rows: 1

   * - Name
     - Description
   * - ``eventname``
     - Name of the event
   * - ``actionname``
     - Name of the action

**Response**

``Expected Response Code: 200``

.. code-block:: json

    {
        "groupScore": {
            "score": 21,
            "group": {
                "id": 6,
                "name": "A"
            }
        }
    }

> The response will indicate the success of dividing the points of the specified Point Group for the Contact by the specified value.

Reviewer: ```suggestion
The response indicates the success of dividing the Points of the specified Point Group for the Contact by the specified value.
```

.. vale off

> Multiply Contact Point Group Score

Reviewer: ```suggestion
Multiply Contact Point Group score
```

**********************************

.. vale on

> Multiply the points of a specific Point Group for a contact by a value.

Reviewer: ```suggestion
Multiply the Points of a specific Point Group for a Contact by a value.
```

.. vale off

**HTTP Request**

.. vale on

``POST /api/contacts/{leadId}/points/groups/{groupId}/times/{value}``

**Data Parameters (optional)**

.. list-table::
   :header-rows: 1

   * - Name
     - Description
   * - ``eventname``
     - Name of the event
   * - ``actionname``
     - Name of the action

**Response**

``Expected Response Code: 200``

.. code-block:: json

    {
        "groupScore": {
            "score": 21,
            "group": {
                "id": 6,
                "name": "A"
            }
        }
    }

> The response will indicate the success of multiplying the points of the specified Point Group for the Contact by the specified value.

Reviewer: ```suggestion
The response indicates the success of multiplying the Points of the specified Point Group for the Contact by the specified value.
```
